### PR TITLE
Implement basic cluster info command

### LINKED ISSUE #58: Add a `cluster info` command to aid users in installation preparation
We would like to be able to help users understand what they're running in their clusters prior to onboarding to Jetstack Secure using `jsctl`.

In order to do this, we'd like to have a command that the user can run to get an overview of the Jetstack tooling and cert-manager components in the cluster in YAML.

The proposal is something like this:

```yaml
components: # things 'running' in the cluster which are Jetstack or cert-manager related.
  certManager:
    namespace: cert-manager
    version: v1.9.0
    installationMechanism: Helm
    apiVersion: v1
  venafiEnhancedIssuer:
    namespace: jetstack-secure
    version: v0.0.1
    installationMechanism: Helm

crds: # matching *cert-manager.io or *jetstack.io CRDs
  jetstack.io:
  - installations.operator.jetstack.io
  cert-manager.io:
  - certificaterequests.cert-manager.io   
  - certificates.cert-manager.io
  - challenges.acme.cert-manager.io       
  - clusterissuers.cert-manager.io
  - issuers.cert-manager.io              
  - orders.acme.cert-manager.io

issuers: # a list of all issuers and cluster issuers in the cluster
- kind: VenafiIssuer
  name: my-tpp-issuer
  namespace: my-tpp
- kind: ClusterIssuer
  name: my-ca-issuer

ingresses:
- name: foo
  namespace: foo
  ingressClass: nginx
  issuer: my-tpp-issuer
         
namespaces:
- cert-manager
- jetstack-secure
```

The command might look like this:

```
jsctl cluster info --format=[json|yaml]
```

### PR DESCRIPTION
**Pull request by charlieegan3**
Fixes #58 

This command shows information like the output shown below. It is capable of showing more information, however, this is what it shows on my test cluster.

```
$ jsctl cluster status 
crds:
- name: cert-manager.io
  items:
  - awspcaissuers.awspca.cert-manager.io
  - certificaterequestpolicies.policy.cert-manager.io
  - certificaterequests.cert-manager.io
  - certificates.cert-manager.io
  - challenges.acme.cert-manager.io
  - clusterissuers.cert-manager.io
  - issuers.cert-manager.io
  - orders.acme.cert-manager.io
- name: jetstack.io
  items:
  - googlecasclusterissuers.cas-issuer.jetstack.io
  - googlecasissuers.cas-issuer.jetstack.io
  - installations.operator.jetstack.io
namespaces:
- jetstack-secure
ingress-shim-ingresses:
- name: example
  namespace: default
  certManagerAnnotations:
    cert-manager.io/cluster-issuer: nameOfClusterIssuer
components:
  cert-manager:
    namespace: jetstack-secure
    versions:
      cainjector: v1.9.1
      controller: v1.9.1
      webhook: v1.9.1
  cert-manager-approver-policy:
    namespace: jetstack-secure
    version: v0.4.0
  jetstack-secure-agent:
    namespace: jetstack-secure
    version: v0.1.38
  jetstack-secure-operator:
    namespace: jetstack-secure
    version: v0.0.1-alpha.17
issuers:
- apiVersion: awspca.cert-manager.io/v1beta1
  kind: AWSPCAIssuer
  name: pca-sample
  namespace: jetstack-secure
- apiVersion: cert-manager.io/v1
  kind: ClusterIssuer
  name: cm-cluster-issuer-sample
- apiVersion: cas-issuer.jetstack.io/v1beta1
  kind: GoogleCASIssuer
  name: googlecasissuer-sample
  namespace: jetstack-secure
- apiVersion: cert-manager.io/v1
  kind: Issuer
  name: cm-issuer-sample
  namespace: jetstack-secure
```

### CODE REVIEW

File: internal/kubernetes/crd.go [hunk 1]
```diff
@@ -1,14 +1,16 @@
-package operator
+package kubernetes
 
 import (
 	"context"
 
-	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
```
Comment (irbekrm) [inline internal/kubernetes/crd.go]: a very minor nit: it'd be good to keep the more semantic import names (as we do in all other projects as well)

**Comment by irbekrm** [comment]
I've not looked at the use of generics in this PR yet, but it seems interesting! Keen to make sure we use the same approach accross projects, so I'd be interested to understand how the generic client compares with using unstructured types/client as well

File: internal/kubernetes/crd.go (hunk 1 above)
Comment (charlieegan3) [reply internal/kubernetes/crd.go]: See 6a56b4e

File: internal/kubernetes/clients/issuers.go [hunk 2]
```diff
@@ -224,3 +233,79 @@ func NewAWSPCAClusterIssuerClient(config *rest.Config) (*Generic[*awspca.AWSPCAC
...
+
+// NewVenafiEnhancedClusterIssuerClient returns an instance of a generic client for querying
+// Venafi enhanced cluster issuers
+func NewVenafiEnhancedClusterIssuerClient(config *rest.Config) (*Generic[*veiapi.VenafiClusterIssuer, *veiapi.VenafiClusterIssuerList], error) {
+	genericClient, err := NewGenericClient[*veiapi.VenafiClusterIssuer, *veiapi.VenafiClusterIssuerList](
+		&GenericClientOptions{
+			RestConfig: config,
+			APIPath:    "/apis",
+			Group:      veiapi.SchemeGroupVersion.Group,
+			Version:    veiapi.SchemeGroupVersion.Group,
```
Comment (irbekrm) [inline internal/kubernetes/clients/issuers.go]: Here should be `veiapi.SchemeGroupVersion.Version`

File: internal/kubernetes/clients/issuers.go [hunk 3]
```diff
@@ -224,3 +233,79 @@ func NewAWSPCAClusterIssuerClient(config *rest.Config) (*Generic[*awspca.AWSPCAC
...
+
+// NewVenafiEnhancedIssuerClient returns an instance of a generic client for querying
+// Venafi enhanced issuers
+func NewVenafiEnhancedIssuerClient(config *rest.Config) (*Generic[*veiapi.VenafiIssuer, *veiapi.VenafiIssuerList], error) {
+	genericClient, err := NewGenericClient[*veiapi.VenafiIssuer, *veiapi.VenafiIssuerList](
+		&GenericClientOptions{
+			RestConfig: config,
+			APIPath:    "/apis",
+			Group:      veiapi.SchemeGroupVersion.Group,
+			Version:    veiapi.SchemeGroupVersion.Group,
```
Comment (irbekrm) [inline internal/kubernetes/clients/issuers.go]: Here should be `veiapi.SchemeGroupVersion.Version`

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thanks Charlie, this is really great work!
I've left two comments for two typos - once i corrected those I was able to run the new command againt a cluster with operator-installed components and get the following output:

```
irbe@jsctl$ go run main.go cluster status
crds:
- name: cert-manager.io
  items:
  - bundles.trust.cert-manager.io
  - certificaterequestpolicies.policy.cert-manager.io
  - certificaterequests.cert-manager.io
  - certificates.cert-manager.io
  - challenges.acme.cert-manager.io
  - clusterissuers.cert-manager.io
  - issuers.cert-manager.io
  - orders.acme.cert-manager.io
- name: jetstack.io
  items:
  - installations.operator.jetstack.io
  - venaficlusterissuers.jetstack.io
  - venafiissuers.jetstack.io
namespaces:
- jetstack-secure
ingress-shim-ingresses: []
components:
  cert-discovery-venafi:
    namespace: jetstack-secure
    version: v0.2.0
  cert-manager:
    namespace: jetstack-secure
    versions:
      cainjector: v1.10.0
      controller: v1.10.0
      webhook: v1.10.0
  cert-manager-approver-policy:
    namespace: jetstack-secure
    version: v0.4.0
  cert-manager-csi-driver:
    namespace: jetstack-secure
    version: v0.1.0
  cert-manager-csi-driver-spiffe:
    namespace: jetstack-secure
    versions:
      approver: v0.2.0
      csi-driver: v0.1.0
  jetstack-secure-operator:
    namespace: jetstack-secure
    version: 950972ef3321ff94c48bd3effb48c7e4548004a7
  trust-manager:
    namespace: jetstack-secure
    version: v0.3.0
  venafi-enhanced-issuer:
    namespace: jetstack-secure
    version: v0.1.4
  venafi-oauth-helper:
    namespace: jetstack-secure
    version: v0.3.0
issuers:
- apiVersion: cert-manager.io/v1
  kind: ClusterIssuer
  name: foo-ca
- apiVersion: cert-manager.io/v1
  kind: ClusterIssuer
  name: spiffe-ca
- apiVersion: cert-manager.io/v1
  kind: Issuer
  name: ca-for-foo-ca
  namespace: jetstack-secure
- apiVersion: cert-manager.io/v1
  kind: Issuer
  name: ca-for-spiffe-ca
  namespace: jetstack-secure
- apiVersion: cert-manager.io/v1
  kind: Issuer
  name: trust-manager
  namespace: jetstack-secure
```

I think we will also want to indicate the installation mechanism (operator in this case) and list policies. However, not in this PR- this looks good to go in for me except for the two typos!

File: internal/kubernetes/clients/issuers.go (hunk 2 above)
Comment (charlieegan3) [reply internal/kubernetes/clients/issuers.go]: Thanks, good spot, addressed in a3bec0050df2f0eacb355dbab4145c71e0a8b8bf

File: internal/kubernetes/clients/issuers.go (hunk 3 above)
Comment (charlieegan3) [reply internal/kubernetes/clients/issuers.go]: Thanks, see a3bec0050df2f0eacb355dbab4145c71e0a8b8bf

**Review by irbekrm — APPROVED** [review]
Thanks a lot for all the good work on this Charlie!

/lgtm

**Comment by charlieegan3** [comment]
🎉 thanks for all the reviews!

**Pull request merged**